packages/backend/src/controllers/house/house.util.ts

```diff
@@ -1,11 +1,55 @@
 // Generate random house code
 import { Injectable } from '@nestjs/common';
 import { randomInt } from 'crypto';
+import { Task, TaskDocument } from 'src/db/task/task.schema';
+
+const maxSafe = 2 ** 48 - 1; //Max int that can be safely reached by randomInt
 
 @Injectable()
 export class HouseUtil {
   generateString(length) {
-    const max_safe = 2 ** 48 - 1; //Max int that can be safely reached by randomInt
-    return randomInt(max_safe).toString(36).substr(1, length).toUpperCase();
+    return randomInt(maxSafe).toString(36).substr(1, length).toUpperCase();
+  }
+
+  selectRandomUser(pool) {
+    if (pool.length == 1) {
+      return pool[0];
+    }
```

Comment:
might not need these lines?

Reply:
`randomInt` throws an error when you have `randomInt(0, 0)` which is the case when there is only one user in the pool

Reply:
I just searched it and it's because the second argument is excluded (first is included)

Reply:
I think that also means the other code in this function is bugged and won't ever return the last user

Reply:
from a quick test `randomInt(0, 1)` works as expected

Reply:
But if it outputs 1, then that is an issue because there is only one element in the array. The arguments and min and max inclusive

Reply:
https://nodejs.org/api/crypto.html#cryptorandomintmin-max-callback

min is inclusive (also not needed because it defaults to 0) and max is exclusive

`randomInt(0, 1)` will always return `0`

Reply:
Ah okay my bad